src/UserSpaceInstrumentation/InjectLibraryInTracee.cpp

```diff
@@ -173,9 +158,7 @@ ErrorMessageOr<uint64_t> FindFunctionAddressWithFallback(pid_t pid, std::string_
                                                                kLibcSoname, kDlcloseInLibc));
 
   // Allocate small memory area in the tracee.
-  OUTCOME_TRY(address, AllocateInTracee(pid, 0, kCodeScratchPadSize));
-  orbit_base::unique_resource code_address{
-      address, [&pid](uint64_t address) { FreeMemoryOrDie(pid, address, kCodeScratchPadSize); }};
+  OUTCOME_TRY(code_address, AllocateInTraceeAsUniqueResource(pid, 0, kCodeScratchPadSize));
 
   // We want to do the following in the tracee:
   // dlclose(handle);
```

Comment:
it is probably because we have implicit cast in `unique_resource<>`, but I think we should remove it and change line 181 to explicitly call `code_address.get()`. Otherwise it is unclear while reading the line 181 where does ownership belong, it is not transferred right?

Reply:
Right, ownership stays with this instance of the unique_resource.

I removed all the `.get()`'s when I found out about the implicit cast. I'm also fine with the more explicit / verbose version. No real preference here.